validator/sawtooth_validator/server/cli.py

```diff
@@ -47,43 +56,41 @@ def parse_args(args):
     return parser.parse_args(args)
 
 
-def ensure_directory(sawtooth_home_path, posix_fallback_path):
-    """Ensures the one of the given sets of directories exists.
-
-    The dirctory in sawtooth_home_path is ensured to exist, if `SAWTOOTH_HOME`
-    exists. If the host operating system is windows, `SAWTOOTH_HOME` is
-    defaulted to `C:\\Program Files (x86)\\Intel\\sawtooth-validator`,
-    Otherwise, the given posix fallback path is ensured to exist.
+def check_directory(path, human_readable_name):
+    """Verify that the directory exists and is readable and writable.
 
     Args:
-        sawtooth_home_dirs (str): Subdirectory of `SAWTOOTH_HOME`.
-        posix_fallback_dir (str): Fallback directory path if `SAWTOOTH_HOME` is
-            unset on posix host system.
+        path (str): a directory which should exist and be writable
+        human_readable_name (str): a human readable string for the directory
+            which is used in logging statements
 
     Returns:
-        str: The path determined to exist.
+        bool: True if an error exists, False otherwise.
```

Comment:
This feels backwards to me.

"Verify that the directory exists and is readable and writable." - but it returns False if everything is ok?